### PR TITLE
test(installer): Add E2E test for proxy setup

### PR DESCRIPTION
**Pull request by BaptisteFoy**
### What does this PR do?
Follow-up of https://github.com/DataDog/datadog-agent/pull/31313 to add an E2E test

### Describe how to test/QA your changes
Only adding a test

### CODE REVIEW

File: test/new-e2e/tests/installer/unix/upgrade_scenario_test.go [hunk 1]
```diff
@@ -600,6 +600,37 @@ func (s *upgradeScenarioSuite) TestUpgradeConfigFailure() {
 	s.mustStopExperiment(datadogAgent)
 }
 
+func (s *upgradeScenarioSuite) TestUpgradeWithProxy() {
+	if s.Env().RemoteHost.OSFlavor == e2eos.Fedora || s.Env().RemoteHost.OSFlavor == e2eos.RedHat {
+		s.T().Skip("Fedora & RedHat can't start the Squid proxy")
+	}
+
+	s.RunInstallScript("DD_REMOTE_UPDATES=true") // No proxy during install, to avoid setting up the APT proxy
```
Comment (arbll) [inline test/new-e2e/tests/installer/unix/upgrade_scenario_test.go:608]: Not sure I follow why APT wouldn't work here ?

File: test/new-e2e/tests/installer/unix/upgrade_scenario_test.go [hunk 2]
```diff
@@ -600,6 +600,37 @@ func (s *upgradeScenarioSuite) TestUpgradeConfigFailure() {
...
+	defer s.Purge()
+	s.host.AssertPackageInstalledByInstaller("datadog-agent")
+
+	// Set proxy config
+	s.Env().RemoteHost.MustExecute(`printf "proxy:\n  http: http://localhost:3128\n  https: http://localhost:3128" | sudo tee -a /etc/datadog-agent/datadog.yaml`)
+	defer func() {
+		s.Env().RemoteHost.MustExecute(`sudo sed -i '/proxy:/,/https:/d' /etc/datadog-agent/datadog.yaml`)
+	}()
```
[the remark this answers is not in the document]
Comment (BaptisteFoy) [reply test/new-e2e/tests/installer/unix/upgrade_scenario_test.go:616]: Yep, the install script doesn't update `datadog.yaml` 😭

File: test/new-e2e/tests/installer/unix/upgrade_scenario_test.go (hunk 1 above)
Comment (BaptisteFoy) [reply test/new-e2e/tests/installer/unix/upgrade_scenario_test.go:608]: It's a bit too hard to support everything that's in ansible / the install script with env vars at the moment. We should add support when we simplify the install methods

**Review by arbll — APPROVED** [review]

**Pull request merged**